test/functional/sharding_connection_tests.js

```diff
@@ -58,6 +58,13 @@ describe('Sharding (Connection)', function() {
     // The actual test we wish to run
     test: function(done) {
       const configuration = this.configuration;
+      if (configuration.usingUnifiedTopology()) {
+        // the unified topology will attempt to connect to all provided servers if you
```

Comment:
I'm not sure I understand. How will the unified topology satisfy the requirements of this test?

Reply:
This test is trying to prove that an invalid mongos in a seedlist will not crash an application. The legacy topologies do most of this resolution during the `connect`, but the unified topology ideally would not connect until the moment it needs to connect.